Feature: Completely Feature Flag Discourse Integration

ardelato opened the pull request:
## Description

This utilizes the pre-existing Discourse integration feature flag to completely toggle all Discourse related components in the codebase.

qa_req 0 This can be viewed in the test cluster instance.

ardelato committed:
feat(discourse): disable Discourse integration and update related components

Set FEATURE__DISCOURSE_INTEGRATION to false in the .env.base file to disable
Discourse integration.

Then thoroughly update the codebase to feature flag all Discourse
related code.

ardelato commented:
deploy_block ☁️ on #10 going out first

ardelato commented:
dev_block ⚡ looks like some Discourse components are still rendering. I need to validate if its due to docker image build or the helm chart configuration.

ardelato commented:
un_dev_block ⚡ The assets were cached. Using an incognito browser rendered the page correctly.

|Before|After|
|:---:|:---:|
|  ![image](https://github.com/user-attachments/assets/beb87c4e-0eef-4532-80db-a174da955b43) |   ![image](https://github.com/user-attachments/assets/dc7dbd2a-2e57-4579-914b-b6d725cd29c1)   |

ardelato commented:
un_deploy_block ⚡ #10 is out

djmetzle commented on app/Http/Controllers/API/DiscourseController.php:
```diff
@@ -19,20 +19,25 @@ public function discussionTopics(Request $request, DiscourseService $discourseSe
     {
         $topics = [];
 
+        if (!config('restarters.features.discourse_integration')) {
+            return response()->json([
+                'success' => 'success',
+                'topics' => $topics
+            ], 200);
```
This seems odd as a stub? Maybe a 204 (No Content) or a 400 code could be more appropriate? (not a blocker, just asking for my own understanding, there's nothing about it in the commit message or description)

ardelato replied:
Addressed in [db4efd3](https://github.com/iFixit/restarters/pull/11/commits/db4efd376e1cd6663a981e9aa1a8a2fa5b8e4a48)

djmetzle commented on routes/web.php:
```diff
@@ -106,147 +106,275 @@
 
     Route::get('workbench', function() {
-        return redirect('https://talk.restarters.net/t/our-work-on-repair-data/1150');
+        if (config('restarters.features.discourse_integration')) {
+            return redirect('https://talk.restarters.net/t/our-work-on-repair-data/1150');
+        } else {
+            return redirect()->route('home');
```
This seems fine, but we can handle this in a bit less of a messy or duplicative way.

First question, do we need to set the 307 back to the homepage? Should these routes not just 404?

Also, we can split out a separate file of routes, and import them wholesale, and conditionally. That could significantly cut down the noise.

ardelato replied:
Addressed in [35dcd05](https://github.com/iFixit/restarters/pull/11/commits/35dcd0538a08a2a55d53d75d9ce4fd9d16fdf684)

We can actually just drop this whole block.

djmetzle reviewed (approved):
CR :+1: with some comments @ardelato, but this looks good to go as-is, if we need it out ASAP.

ardelato committed:
fix(api): return more appropriate response for stubbed out discourse integration

ardelato committed:
chore(routes): remove leftover workbench route

The Workbench functionality was removed in https://github.com/TheRestartProject/restarters.net/pull/787
but there was a leftover route in routes/web.php to handle the
case of accessing the page directly. That said, we don't need to
support the backwards compatibility so we can just remove it.

mlahargou reviewed (approved):
CR 📱 on the changes since Metz's CR.
